service/resource/vaultcrt/desired.go

```diff
@@ -0,0 +1,43 @@
+package vaultcrt
+
+import (
+	"context"
+
+	"github.com/giantswarm/certificatetpr"
+	"github.com/giantswarm/microerror"
+	apismetav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	apiv1 "k8s.io/client-go/pkg/api/v1"
+
+	"github.com/giantswarm/cert-operator/service/key"
+)
+
+func (r *Resource) GetDesiredState(ctx context.Context, obj interface{}) (interface{}, error) {
+	customObject, err := key.ToCustomObject(obj)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	r.logger.Log("cluster", key.ClusterID(customObject), "debug", "computing the desired secret")
+
+	// NOTE that the actual secret content here is left blank because only the
+	// issuer backend, e.g. Vault, can generate certificates. This has to be
+	// considered when computing the create, delete and update state.
```

Comment:
So, desired state is the empty string, and we add in the certificates when we actually create them?

Reply:
Yes, reason is that we do not want to generate certificates every five minutes to not use them at the end. This is just due to the fact how certificate generation works in general.